diff --git a/node_modules/react-scripts/config/webpack.config.js b/node_modules/react-scripts/config/webpack.config.js
index e465d8e..0df1b2b 100644
--- a/node_modules/react-scripts/config/webpack.config.js
+++ b/node_modules/react-scripts/config/webpack.config.js
@@ -95,6 +95,9 @@ const hasJsxRuntime = (() => {
   }
 })();
 
+const buildNumber = process.env.BUILD_NUMBER;
+const staticPath = `static${buildNumber ? `-${buildNumber}` : ''}`;
+
 // This is the production and development configuration.
 // It is focused on developer experience, fast rebuilds, and a minimal bundle.
 module.exports = function (webpackEnv) {
@@ -219,13 +222,15 @@ module.exports = function (webpackEnv) {
       // There will be one main bundle, and one file per asynchronous chunk.
       // In development, it does not produce real files.
       filename: isEnvProduction
-        ? 'static/js/[name].[contenthash:8].js'
+        ? `${staticPath}/js/[name].[contenthash:8].js`
         : isEnvDevelopment && 'static/js/bundle.js',
       // There are also additional JS chunk files if you use code splitting.
       chunkFilename: isEnvProduction
-        ? 'static/js/[name].[contenthash:8].chunk.js'
+        ? `${staticPath}/js/[name].[contenthash:8].chunk.js`
         : isEnvDevelopment && 'static/js/[name].chunk.js',
-      assetModuleFilename: 'static/media/[name].[hash][ext]',
+      assetModuleFilename: isEnvProduction
+        ? `${staticPath}/media/[name].[hash].[ext]`
+        : 'static/media/[name].[hash].[ext]',
       // webpack uses `publicPath` to determine where the app is being served from.
       // It requires a trailing slash, or the file assets will get an incorrect path.
       // We inferred the "public path" (such as / or /my-project) from homepage.
@@ -403,7 +408,9 @@ module.exports = function (webpackEnv) {
                 {
                   loader: require.resolve('file-loader'),
                   options: {
-                    name: 'static/media/[name].[hash].[ext]',
+                    name: isEnvProduction
+                      ? `${staticPath}/media/[name].[hash].[ext]`
+                      : 'static/media/[name].[hash].[ext]',
                   },
                 },
               ],
@@ -666,8 +673,8 @@ module.exports = function (webpackEnv) {
         new MiniCssExtractPlugin({
           // Options similar to the same options in webpackOptions.output
           // both options are optional
-          filename: 'static/css/[name].[contenthash:8].css',
-          chunkFilename: 'static/css/[name].[contenthash:8].chunk.css',
+          filename: isEnvProduction ? `${staticPath}/css/[name].[contenthash:8].css` : 'static/css/[name].[contenthash:8].css',
+          chunkFilename: isEnvProduction ? `${staticPath}/css/[name].[contenthash:8].chunk.css` : 'static/css/[name].[contenthash:8].chunk.css',
         }),
       // Generate an asset manifest file with the following content:
       // - "files" key: Mapping of all asset filenames to their corresponding
